### PR TITLE
Add option for supporting partial program input to clutz.

### PR DESCRIPTION
**Pull request by rkirov**
When the flag is set, clutz will use setAssumeForwardDeclaration for all
unknown types, and not error on incomplete inputs.

Adds /partial/ directory for tests that run with that option on.

### CODE REVIEW

File: src/test/java/com/google/javascript/clutz/ProgramSubject.java [hunk 1]
```diff
@@ -105,7 +109,7 @@ StringSubject diagnosticStream() {
       roots.add("main.js");
     }
 
-    List<SourceFile> externFiles = new ArrayList<>();
+    List<SourceFile> externFiles;
```
Comment (evmar) [inline src/test/java/com/google/javascript/clutz/ProgramSubject.java]: This seems unrelated (?)

Harmless if so, but weird if unintentional

File: src/main/java/com/google/javascript/clutz/Options.java [hunk 2]
```diff
@@ -71,6 +71,14 @@
   )
   List<String> entryPoints = new ArrayList<>();
 
+  @Option(
+    name = "--partialInput",
+    usage =
+        "allow input of incomplete programs. All unknown types will be treated as forward"
+            + "declared."
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/Options.java]: nit: you're missing a space here in the string.

File: src/test/java/com/google/javascript/clutz/partial/missing_type.d.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+declare namespace ಠ_ಠ.clutz.module$exports$missing$type {
+  var x : any ;
```
Comment (mprobst) [inline src/test/java/com/google/javascript/clutz/partial/missing_type.d.ts]: This should end up being `var x : Missing ;` eventually, but needs further changes, right?

File: src/test/java/com/google/javascript/clutz/ProgramSubject.java (hunk 1 above)
Comment (rkirov) [inline src/test/java/com/google/javascript/clutz/ProgramSubject.java]: yep, unrelated clean-up. Intellij made is different color, so hard to resists :)

File: src/test/java/com/google/javascript/clutz/partial/missing_type.d.ts (hunk 3 above)
Comment (rkirov) [inline src/test/java/com/google/javascript/clutz/partial/missing_type.d.ts]: yep, one PR at a time. I am unrolling a bunch of local changes I have internally. That would be the next PR.
